runnumber-rest/api.py

```diff
@@ -0,0 +1,115 @@
+import os
+from datetime import datetime
+import flask
+from flask_restful import Api, Resource
+from flask_sqlalchemy import SQLAlchemy
+from sqlalchemy import func
+
+__all__ = ["app", "api", "db"]
+
+app = flask.Flask(__name__)
+
+app.config["SQLALCHEMY_DATABASE_URI"] = os.environ.get(
+    "DATABASE_URI", "sqlite:////tmp/test.sqlite"
+)
+
+db = SQLAlchemy(app)
+api = Api(app)
+
+from database import RunNumber
+from authentication import auth
+
+# $ curl -u fooUsr:barPass -X GET np04-srv-021:5000//runnumber/get
+@api.resource("/runnumber/get")
+class getRunNumber(Resource):
+    """
+    returns the run number of the previous run
+    if no previous run number, returns in format: Null
+    otherwise returns the last run number in format: [1000]
+    
+    """
+
+    @auth.login_required
+    def get(self):
+        rowRes = []
+        try:
+            rowRes.append(
+                db.session.execute(db.select(func.max(RunNumber.rn))).scalar_one()
+            )
+        except Exception as e:
+            (err_obj,) = e.args
+            print("Exception:", err_obj.message)
+            resp = flask.make_response(flask.jsonify({"Exception": err_obj.message}))
+            return resp
+        print(f"getRunNumber: result {rowRes}")
+        resp = flask.make_response(flask.jsonify(rowRes))
+        return resp
+
+# $ curl -u fooUsr:barPass -X GET np04-srv-021:5000//runnumber/getnew
+@api.resource("/runnumber/getnew")
+class getNewtRunNumber(Resource):
+    """
+    create a new run in the database with a new run number which is previous run number +1
+    if no previous run number, returns: [1000]
+    otherwise return the run number in format: [1001]
+    """
+
+    @auth.login_required
+    def get(self):
+        rowRes = []
+        try:
+            # if we start at a higher number
+            # the primary key sequence may not match
+            current_max_run = db.session.execute(
+                db.select(func.max(RunNumber.rn))
+            ).scalar_one()
+            if current_max_run is None:
+                current_max_run = 1000
```

Comment:
The `1000` here should probably be controlled by an environment variable in a similar way to how `DEBUG` is checked in the application startup code.

Reply:
But if its controlled by an environment variable then surely that defeats the purpose of the run number being set automatically?

Reply:
In the previous version of runservices we just artificially introduced a fake run so all the other runs could iterate up from it

Reply:
Happy to change it either way though

Reply:
I was thinking something along the lines of:

```
                current_max_run = int(os.getenv("RUN_START", "1000"))
```

Reply:
Added it in latest commit